pkg/querier/blocks_store_queryable.go

```diff
@@ -1116,6 +1116,8 @@ func isRetryableError(err error) bool {
 	switch status.Code(err) {
 	case codes.Unavailable:
 		return true
+	case codes.ResourceExhausted:
+		return errors.Is(err, storegateway.ErrTooManyInflightRequests)
```

Comment:
Do we need to decrease retry count in this case if it is failed due to too many inflight requests?
Otherwise it seems like a behavior change. Requests that queued up before might succeed before but now it might fail

Reply:
I treated this change as a simple "fail fast from SG". I thought of manually incrementing the retry count, but I thought it's also a change in behavior of we start additing exceptions to the retry count (ie. we retry more than 3 times in certain cases). I'm curious to get comments from others about this.

Reply:
> it could also be a change in behavior from the perspective of having different retry counts for different scenarios
I feel this is fine... As we fail fast at SG so it hasn't started the expensive query execution.

I am also ok if we start simple and iterate on top of it
Thoughts? @alanprot @alvinlin123 @friedrichg

Reply:
I am fine with this